sled-hardware/src/lib.rs

```diff
@@ -73,48 +73,76 @@ pub enum SledMode {
 }
 
 /// Describes properties that should uniquely identify a Gimlet.
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
-pub struct Baseboard {
-    identifier: String,
-    model: String,
-    revision: i64,
+#[derive(
+    Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize, JsonSchema,
+)]
+pub enum Baseboard {
+    Gimlet { identifier: String, model: String, revision: i64 },
+
+    Unknown,
+
+    Pc { identifier: String, model: String, revision: i64 },
 }
```

Comment:
It's not clear to me why it's worth diverging from other implementations of baseboard when both arms of the enum have the same shape?

Reply:
Really, only to preserve the type information for later pattern matching. I could have hard coded `Pc`'s return values in the model and revision impl functions for Baseboard but thought it would make more sense as configurable fields.